src/devices/src/virtio/block/device.rs

```diff
@@ -180,7 +180,8 @@ impl DiskProperties {
         // The config space is little endian.
         let mut config = Vec::with_capacity(CONFIG_SPACE_SIZE);
         for i in 0..CONFIG_SPACE_SIZE {
-            config.push(u8::try_from(self.nsectors >> (8 * i)).unwrap());
+            #[allow(clippy::cast_possible_truncation)]
+            config.push((self.nsectors >> (8 * i)) as u8);
         }
         config
```

Comment:
`as` only wraps in release mode, in debug it will panic. How do you know this conversion needs to wrap?

Reply:
My understanding of this code is that it's writing `CONFIG_SPACE_SIZE` bytes of `self.nsectors` into `config`. When `i = 0`, only the least-significant byte is added; when `i=1`, the next byte is added.

The original version of this code also used `as u8` (I changed it in my original commit but undid it here), so I don't see any reason why there should be any new panics in this part of the code.

Reply:
https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast

> Casting from a larger integer to a smaller integer (e.g. u32 -> u8) will truncate

Apologies I was incorrect, for now this is fine.

Reply:
I only saw this just now; I've added that comment.